README.md
<!-- SPDX-License-Identifier: MIT -->
# ![Product Kit Tailwind Logo](./docs/images/pk_tailwind_title_image.png)

![license](https://img.shields.io/badge/license-MIT-38de03e?style=flat)
[![npm](https://img.shields.io/npm/v/@mercedes-benz/productkit-tailwind)](https://www.npmjs.com/package/@mercedes-benz/productkit-tailwind)

> **Note** <br>
> With the release of v2.2.0 and all future versions, product-kit_tailwind is available under the npm scope @mercedes-benz/productkit-tailwind.

Product Kit Tailwind translates design tokens of [Product Kit Core](https://github.com/mercedes-benz/product-kit_core) in the form of a predefined tailwind.config.js which can be imported in your own Tailwind config so you can start writing frontend code without learning how to adapt to styleguide specifications.

The following theme definitions have predefined values from [Product Kit Core](https://github.com/mercedes-benz/product-kit_core):

* Colors
* Typography
* Spacing
* Layout
* Opacity
* Sizes

> If you need further improvements to Product Kit Tailwind or miss something from the styleguide feel free to open an [issue](https://github.com/mercedes-benz/product-kit_tailwind/issues)!

## Installation

`1.` Add Product Kit Tailwind as a dependency through npm:

```bash
npm install -D @mercedes-benz/productkit-tailwind
```

`2.` Import Product Kit Tailwind to your tailwind.config.js like so:

```javascript
module.exports = {
  presets: [
    require('@mercedes-benz/productkit-tailwind/src/productkit.tailwind.config.js')
  ]
```

`3.` Add additional style information to your main `styles` css

```css
@import 'tailwindcss/base';
@import '@mercedes-benz/productkit-tailwind/src/styles.css';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
```

## Usage

Product Kit Tailwind offers utility functions for frontends incorporating the styleguide of Mercedes-Benz Tech Innovation.

If you want to get the most out of Product Kit Tailwind please familiarise yourself with [Product Kit Core](https://github.com/mercedes-benz/product-kit_core) and its design tokens.

### Colors

All colors defined in [Product Kit Core](https://github.com/mercedes-benz/product-kit_core) can be referenced through Tailwinds utility functions:

```html
<div class="bg-primary text-contrast-background border-goldentainoi-400">
```

There is no need to specify dark mode specific colors. Application colors like `primary`, `secondary`, `background` or `surface` will automatically switch to their dark mode representations. Learn more about this behavior in the `Dark Mode` section below.

> If you want to have a list of all the predefined colors please take a look at the color defnitions [here](https://github.com/mercedes-benz/product-kit_tailwind/raw/main/src/theme/color/colors.js).

### Spacings

Product Kit Tailwind comes preloaded with styleguide specific spacing definitions which can be used through Tailwinds utility functions:

```html
<div class="mt-2 px-3 gap-y-16">
```

Additionally to these abolute spacing defintions you can use opinionated spacing styles that behave according to viewport sizes (dynamic spacing):

```html
<div class="mt-m">
```

The class above will translate to the following css:

```css
...
@media (min-width: 600px) {
    .mt-m {
        margin-top: 32px;
    }
}
@media (min-width: 905px) {
    .mt-m {
        margin-top: 48px;
    }
}
...
```

### Typography

Every text style defined in [Product Kit Core](https://github.com/mercedes-benz/product-kit_core) can easily be specified with a css class:

```html
<span class="text-headline3">
    This is a headline!
</span>
```

When using a predefined text style class you get responsive behavior for free! The above class definition will translate to: (example values)

```css
.text-headline3 {
    font-size: 34px;
    line-height: 44px;
    letter-spacing: 0.1px;
    text-transform: uppercase;
}
@screen m {
    .text-headline3 {
        font-size: 48px;
        line-height: 60px;
        letter-spacing: 0px;
    }
}
```

> If you want to have a list of all the predefined text styles please take a look at the responsive typography plugin [here](https://github.com/mercedes-benz/product-kit_tailwind/raw/main/src/plugins/responsive-typography/index.js).

#### Headline Divider

The styleguide defines a divider below headlines. You can use the predefined class `.divider` to get a responsive line below headlines.

```html
<div class="divider" />
```

> **Note:** You have to calculate the width of the divider according to the headline size. Learn more about the divider behavior in the Mercedes-Benz Tech Innovation styleguide.

### Responsive Layout

Product Kit Tailwind provides a `.container` and `.container-wide` class that you can use to adapt the content of your frontend to layout specifications of [Product Kit Core](https://github.com/mercedes-benz/product-kit_core). Everything inside your container element is arranged according to predefined breakpoints and grid definitions.

```html
<div class="container">
    Responsive Body
</div>
```

The class above will translate to the following css: (example values)

```css
.container {
    width: 90%;
    display: grid;
    margin: auto;
}
@screen xs {
    .container {
        width: 87%;
        column-gap: 16px;
        grid-template-columns: repeat(4, minmax(0, 1fr));
    }
}
...
```

### Dark Mode

You can get dark mode specific colors by adding `.dark` to the root element of your html code:

```html
<html class="dark">
```

This will automatically switch application colors to their dark mode counterpart.

If you need further guidance in how to implement a dark mode switch please take a look inside the test directory [here](https://github.com/mercedes-benz/product-kit_tailwind/raw/main/test/public/index.html).

The following color names will switch automatically:

* primary
* secondary
* tertiary
* quternary
* background
* surface
* outline
* error
* success
* **all contrast colors**

> **Note**: Contrast colors are used as text colors. If text is displayed on the background color `primary` the text color should be `contrast-primary`. Please also make yourself familiar with text opacity values. You can find further information [here](https://github.com/mercedes-benz/product-kit_core/blob/main/src/tokens/opacity/application.json).

## Contributing

We welcome any contributions.
If you want to contribute to this project, please read the [contributing guide](CONTRIBUTING.md).

## Code of Conduct

Please read our [Code of Conduct](https://github.com/mercedes-benz/foss/blob/master/CODE_OF_CONDUCT.md) as it is our base for interaction.

## License

This project is licensed under the [MIT LICENSE](LICENSE).

## Provider Information

Please visit <https://www.mercedes-benz-techinnovation.com/en/imprint/> for information on the provider.

Notice: Before you use the program in productive use, please take all necessary precautions,
e.g. testing and verifying the program with regard to your specific use.
The program was tested solely for our own use cases, which might differ from yours.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.

## [2.2.0]

### Changed
- Update to Product Kit Core 2.2.1

## [2.1.0]

### Added
- Added new typography tokens

### Changed
- Updated to Product Kit Core 2.1.0

## [2.0.1]

### Fixed
- Fixed opacity tokens

## [2.0.0]

### Changed
- Updated Product Kit Core dependency to 2.0.0
- Updated styles according to Product Kit Core v2

### Removed
- Removed styles not available in Product Kit Core v2 anymore

## [1.1.0]

### Added
- Added `.container-wide` class for wide application layouts

### Changed
- Changed main entry point to the `src/` folder
- Updated Product Kit Core dependency to 1.3.0



tailwind.config.js

const brandColors = {
  white: "var(--color-white)",
  black: "var(--color-black)",
  neutral: {
    50: 'var(--color-brand-neutral-50)',
    100: 'var(--color-brand-neutral-100)',
    200: 'var(--color-brand-neutral-200)',
    300: 'var(--color-brand-neutral-300)',
    400: 'var(--color-brand-neutral-400)',
    500: 'var(--color-brand-neutral-500)',
    600: 'var(--color-brand-neutral-600)',
    700: 'var(--color-brand-neutral-700)',
    800: 'var(--color-brand-neutral-800)',
    900: 'var(--color-brand-neutral-900)',
  },
  blue: {
    50: 'var(--color-brand-blue-50)',
    100: 'var(--color-brand-blue-100)',
    200: 'var(--color-brand-blue-200)',
    300: 'var(--color-brand-blue-300)',
    400: 'var(--color-brand-blue-400)',
    500: 'var(--color-brand-blue-500)',
    600: 'var(--color-brand-blue-600)',
    700: 'var(--color-brand-blue-700)',
    800: 'var(--color-brand-blue-800)',
    900: 'var(--color-brand-blue-900)',
  },
  deepblue: {
    50: 'var(--color-brand-deepblue-50)',
    100: 'var(--color-brand-deepblue-100)',
    200: 'var(--color-brand-deepblue-200)',
    300: 'var(--color-brand-deepblue-300)',
    400: 'var(--color-brand-deepblue-400)',
    500: 'var(--color-brand-deepblue-500)',
    600: 'var(--color-brand-deepblue-600)',
    700: 'var(--color-brand-deepblue-700)',
    800: 'var(--color-brand-deepblue-800)',
    900: 'var(--color-brand-deepblue-900)',
  },
  grayblue: {
    50: 'var(--color-brand-grayblue-50)',
    100: 'var(--color-brand-grayblue-100)',
    200: 'var(--color-brand-grayblue-200)',
    300: 'var(--color-brand-grayblue-300)',
    400: 'var(--color-brand-grayblue-400)',
    500: 'var(--color-brand-grayblue-500)',
    600: 'var(--color-brand-grayblue-600)',
    700: 'var(--color-brand-grayblue-700)',
    800: 'var(--color-brand-grayblue-800)',
    900: 'var(--color-brand-grayblue-900)',
  },
  pink: {
    50: 'var(--color-brand-pink-50)',
    100: 'var(--color-brand-pink-100)',
    200: 'var(--color-brand-pink-200)',
    300: 'var(--color-brand-pink-300)',
    400: 'var(--color-brand-pink-400)',
    500: 'var(--color-brand-pink-500)',
    600: 'var(--color-brand-pink-600)',
    700: 'var(--color-brand-pink-700)',
    800: 'var(--color-brand-pink-800)',
    900: 'var(--color-brand-pink-900)',
  },
  purple: {
    50: 'var(--color-brand-purple-50)',
    100: 'var(--color-brand-purple-100)',
    200: 'var(--color-brand-purple-200)',
    300: 'var(--color-brand-purple-300)',
    400: 'var(--color-brand-purple-400)',
    500: 'var(--color-brand-purple-500)',
    600: 'var(--color-brand-purple-600)',
    700: 'var(--color-brand-purple-700)',
    800: 'var(--color-brand-purple-800)',
    900: 'var(--color-brand-purple-900)',
  },
  red: {
    50: 'var(--color-brand-red-50)',
    100: 'var(--color-brand-red-100)',
    200: 'var(--color-brand-red-200)',
    300: 'var(--color-brand-red-300)',
    400: 'var(--color-brand-red-400)',
    500: 'var(--color-brand-red-500)',
    600: 'var(--color-brand-red-600)',
    700: 'var(--color-brand-red-700)',
    800: 'var(--color-brand-red-800)',
    900: 'var(--color-brand-red-900)',
  },
  green: {
    50: 'var(--color-brand-green-50)',
    100: 'var(--color-brand-green-100)',
    200: 'var(--color-brand-green-200)',
    300: 'var(--color-brand-green-300)',
    400: 'var(--color-brand-green-400)',
    500: 'var(--color-brand-green-500)',
    600: 'var(--color-brand-green-600)',
    700: 'var(--color-brand-green-700)',
    800: 'var(--color-brand-green-800)',
    900: 'var(--color-brand-green-900)',
  }
}

const brandSpacing = {
  '0': '0px',
  '0.5': 'var(--size-space-halfx)',
  '1': 'var(--size-space-1x)',
  '2': 'var(--size-space-2x)',
  '3': 'var(--size-space-3x)',
  '4': 'var(--size-space-4x)',
  '6': 'var(--size-space-6x)',
  '8': 'var(--size-space-8x)',
  '9': 'var(--size-space-9x)',
  '10': 'var(--size-space-10x)',
  '11': 'var(--size-space-11x)',
  '12': 'var(--size-space-12x)',
  '16': 'var(--size-space-16x)',
  '24': 'var(--size-space-24x)',
}

module.exports = {
  darkMode: 'class',
  content: [
    './src/**/*.{html,js}',
  ],
  theme: {
    screens: {
      xs: 'var(--size-screen-xs)',
      s: 'var(--size-screen-s)',
      m: 'var(--size-screen-m)',
      l: 'var(--size-screen-l)',
      xl: 'var(--size-screen-xl)',
    },
    spacing: {
      ...brandSpacing,
    },
    responsiveSpacing: {
      xs: {
        '3xs': 'var(--size-application-spacing-xs-3xs)',
        xxs: 'var(--size-application-spacing-xs-xxs)',
        xs: 'var(--size-application-spacing-xs-xs)',
        s: 'var(--size-application-spacing-xs-s)',
        m: 'var(--size-application-spacing-xs-m)',
        l: 'var(--size-application-spacing-xs-l)',
        xl: 'var(--size-application-spacing-xs-xl)',
        xxl: 'var(--size-application-spacing-xs-xxl)',
        '3xl': 'var(--size-application-spacing-xs-3xl)'
      },
      s: {
        '3xs': 'var(--size-application-spacing-s-3xs)',
        xxs: 'var(--size-application-spacing-s-xxs)',
        xs: 'var(--size-application-spacing-s-xs)',
        s: 'var(--size-application-spacing-s-s)',
        m: 'var(--size-application-spacing-s-m)',
        l: 'var(--size-application-spacing-s-l)',
        xl: 'var(--size-application-spacing-s-xl)',
        xxl: 'var(--size-application-spacing-s-xxl)',
        '3xl': 'var(--size-application-spacing-s-3xl)'
      },
      m: {
        '3xs': 'var(--size-application-spacing-m-3xs)',
        xxs: 'var(--size-application-spacing-m-xxs)',
        xs: 'var(--size-application-spacing-m-xs)',
        s: 'var(--size-application-spacing-m-s)',
        m: 'var(--size-application-spacing-m-m)',
        l: 'var(--size-application-spacing-m-l)',
        xl: 'var(--size-application-spacing-m-xl)',
        xxl: 'var(--size-application-spacing-m-xxl)',
        '3xl': 'var(--size-application-spacing-m-3xl)'
      },
      l: {
        '3xs': 'var(--size-application-spacing-l-3xs)',
        xxs: 'var(--size-application-spacing-l-xxs)',
        xs: 'var(--size-application-spacing-l-xs)',
        s: 'var(--size-application-spacing-l-s)',
        m: 'var(--size-application-spacing-l-m)',
        l: 'var(--size-application-spacing-l-l)',
        xl: 'var(--size-application-spacing-l-xl)',
        xxl: 'var(--size-application-spacing-l-xxl)',
        '3xl': 'var(--size-application-spacing-l-3xl)'
      },
      xl: {
        '3xs': 'var(--size-application-spacing-xl-3xs)',
        xxs: 'var(--size-application-spacing-xl-xxs)',
        xs: 'var(--size-application-spacing-xl-xs)',
        s: 'var(--size-application-spacing-xl-s)',
        m: 'var(--size-application-spacing-xl-m)',
        l: 'var(--size-application-spacing-xl-l)',
        xl: 'var(--size-application-spacing-xl-xl)',
        xxl: 'var(--size-application-spacing-xl-xxl)',
        '3xl': 'var(--size-application-spacing-xl-3xl)'
      }
    },
    fontFamily: {
      sans: ['var(--font-family-brand)', 'sans-serif'],
      mono: ['var(--font-family-monospace)', 'sans-serif'],
    },
    fontWeight: {
      light: 'var(--font-weight-light)',
      regular: 'var(--font-weight-regular)',
      bold: 'var(--font-weight-bold)',
      black: 'var(--font-weight-black)',
    },
    fontSize: {
      '3xs': 'var(--size-font-3xs)',
      'xxs': 'var(--size-font-xxs)',
      'xs': 'var(--size-font-xs)',
      's': 'var(--size-font-s)',
      'm': 'var(--size-font-m)',
      'l': 'var(--size-font-l)',
      'xl': 'var(--size-font-xl)',
      'xxl': 'var(--size-font-xxl)',
      '3xl': 'var(--size-font-3xl)',
      '4xl': 'var(--size-font-4xl)',
    },
    lineHeight: {
      '3xs': 'var(--size-line-height-3xs)',
      'xxs': 'var(--size-line-height-xxs)',
      'xs': 'var(--size-line-height-xs)',
      's': 'var(--size-line-height-s)',
      'm': 'var(--size-line-height-m)',
      'l': 'var(--size-line-height-l)',
      'xl': 'var(--size-line-height-xl)',
      'xxl': 'var(--size-line-height-xxl)',
      '3xl': 'var(--size-line-height-3xl)',
    },
    letterSpacing: {
      '3xs': 'var(--size-letter-spacing-3xs)',
      'xxs': 'var(--size-letter-spacing-xxs)',
      'xs': 'var(--size-letter-spacing-xs)',
      's': 'var(--size-letter-spacing-s)',
      'm': 'var(--size-letter-spacing-m)',
      'l': 'var(--size-letter-spacing-l)',
      'xl': 'var(--size-letter-spacing-xl)',
      'xxl': 'var(--size-letter-spacing-xxl)',
      '3xl': 'var(--size-letter-spacing-3xl)',
      '4xl': 'var(--size-letter-spacing-4xl)',
    },
    opacity: {
      '38': 'var(--opacity-brand-1)',
      '60': 'var(--opacity-brand-2)',
      '74': 'var(--opacity-brand-3)',
      '87': 'var(--opacity-brand-4)',
    },
    colors: {
      ...brandColors,
      primary: "var(--color-application-primary)",
      secondary: "var(--color-application-secondary)",
      tertiary: "var(--color-application-tertiary)",
      quaternary: "var(--color-application-quaternary)",
      background: "var(--color-application-background)",
      surface: "var(--color-application-surface)",
      error: "var(--color-application-error)",
      outline: "var(--color-application-outline)",
      contrast: {
        primary: "var(--color-application-contrast-primary)",
        secondary: "var(--color-application-contrast-secondary)",
        tertiary: "var(--color-application-contrast-tertiary)",
        quaternary: "var(--color-application-contrast-quaternary)",
        background: "var(--color-application-contrast-background)",
        surface: "var(--color-application-contrast-surface)",
        error: "var(--color-application-contrast-error)",
        outline: "var(--color-application-contrast-outline)",
      },
    },
    extend: {},
  },
  corePlugins: {
    container: false
  },
  plugins: [
    require('./src/plugins/responsive-spacings'),
    {
      theme: {
        extend: {
          transitionDuration: {
            0: "0ms",
          },
        },
      },
    },
    function ({ addComponents, theme }) {
      addComponents({
        '.container': {
          marginLeft: 'auto',
          marginRight: 'auto',
          width: '100%',
          display: 'grid',
          '@screen xs': {
            maxWidth: '100%',
            paddingLeft: '16px',
            paddingRight: '16px',
            columnGap: '16px',
            gridTemplateColumns: 'repeat(4, minmax(0, 1fr))'
          },
          '@screen s': {
            maxWidth: '100%',
            paddingLeft: '32px',
            paddingRight: '32px',
            columnGap: '16px',
            gridTemplateColumns: 'repeat(8, minmax(0, 1fr))'
          },
          '@screen m': {
            maxWidth: '840px',
            paddingLeft: '0px',
            paddingRight: '0px',
            columnGap: '24px',
            gridTemplateColumns: 'repeat(12, minmax(0, 1fr))'
          },
          '@screen l': {
            maxWidth: '100%',
            paddingLeft: '192px',
            paddingRight: '192px',
            columnGap: '24px',
            gridTemplateColumns: 'repeat(12, minmax(0, 1fr))'
          },
          '@screen xl': {
            maxWidth: '1040px',
            paddingLeft: '0px',
            paddingRight: '0px',
            columnGap: '24px',
            gridTemplateColumns: 'repeat(12, minmax(0, 1fr))'
          },
        },
        '.text-headline1': { 
          fontSize: theme('fontSize.3xl'),
          lineHeight: theme('lineHeight.xxl'),
          letterSpacing: theme('letterSpacing.xxs'),
          textTransform: 'uppercase',
          '@screen m': {
            fontSize: theme('fontSize.4xl'),
            lineHeight: theme('lineHeight.3xl'),
            letterSpacing: theme('letterSpacing.3xs'),
          }
        },
        '.text-headline2': { 
          fontSize: theme('fontSize.xxl'),
          lineHeight: theme('lineHeight.xl'),
          letterSpacing: theme('letterSpacing.xs'),
          textTransform: 'uppercase',
          '@screen m': {
            fontSize: theme('fontSize.3xl'),
            lineHeight: theme('lineHeight.xxl'),
            letterSpacing: theme('letterSpacing.xxs'),
          }
        },
        '.text-headline3': { 
          fontSize: theme('fontSize.xl'),
          lineHeight: theme('lineHeight.l'),
          letterSpacing: theme('letterSpacing.s'),
          textTransform: 'uppercase',
          '@screen m': {
            fontSize: theme('fontSize.xxl'),
            lineHeight: theme('lineHeight.xl'),
            letterSpacing: theme('letterSpacing.xs'),
          }
        },
        '.text-headline4': { 
          fontSize: theme('fontSize.l'),
          lineHeight: theme('lineHeight.m'),
          letterSpacing: theme('letterSpacing.m'),
          '@screen m': {
            fontSize: theme('fontSize.xl'),
            lineHeight: theme('lineHeight.l'),
            letterSpacing: theme('letterSpacing.s'),
          }
        },
        '.text-headline5': { 
          fontSize: theme('fontSize.m'),
          lineHeight: theme('lineHeight.s'),
          letterSpacing: theme('letterSpacing.l'),
          '@screen m': {
            fontSize: theme('fontSize.l'),
            lineHeight: theme('lineHeight.m'),
            letterSpacing: theme('letterSpacing.m'),
          }
        },
        '.text-headline6': { 
          fontSize: theme('fontSize.s'),
          lineHeight: theme('lineHeight.xs'),
          letterSpacing: theme('letterSpacing.xl'),
          '@screen m': {
            fontSize: theme('fontSize.m'),
            lineHeight: theme('lineHeight.s'),
            letterSpacing: theme('letterSpacing.l'),
          }
        },
        '.text-subtitle1': { 
          fontSize: theme('fontSize.xs'),
          lineHeight: theme('lineHeight.s'),
          letterSpacing: theme('letterSpacing.l'),
          '@screen m': {
            fontSize: theme('fontSize.s'),
            lineHeight: theme('lineHeight.m'),
            letterSpacing: theme('letterSpacing.m'),
          }
        },
        '.text-subtitle2': { 
          fontSize: theme('fontSize.xxs'),
          lineHeight: theme('lineHeight.xs'),
          letterSpacing: theme('letterSpacing.m'),
          '@screen m': {
            fontSize: theme('fontSize.xs'),
            lineHeight: theme('lineHeight.s'),
            letterSpacing: theme('letterSpacing.s'),
          }
        },
        '.text-body1': { 
          fontSize: theme('fontSize.xs'),
          lineHeight: theme('lineHeight.xs'),
          letterSpacing: theme('letterSpacing.xxl'),
          '@screen m': {
            fontSize: theme('fontSize.s'),
            lineHeight: theme('lineHeight.s'),
            letterSpacing: theme('letterSpacing.xxl'),
          }
        },
        '.text-body2': { 
          fontSize: theme('fontSize.xxs'),
          lineHeight: theme('lineHeight.xxs'),
          letterSpacing: theme('letterSpacing.xl'),
          '@screen m': {
            fontSize: theme('fontSize.xs'),
            lineHeight: theme('lineHeight.xs'),
            letterSpacing: theme('letterSpacing.l'),
          }
        },
        '.text-button': { 
          fontSize: theme('fontSize.xs'),
          lineHeight: theme('lineHeight.xxs'),
          letterSpacing: theme('letterSpacing.3xl'),
          textTransform: 'uppercase',
          '@screen m': {
            fontSize: theme('fontSize.xs'),
            lineHeight: theme('lineHeight.xxs'),
            letterSpacing: theme('letterSpacing.3xl'),
          }
        },
        '.text-caption': { 
          fontSize: theme('fontSize.3xs'),
          lineHeight: theme('lineHeight.xxs'),
          letterSpacing: theme('letterSpacing.xl'),
          '@screen m': {
            fontSize: theme('fontSize.3xs'),
            lineHeight: theme('lineHeight.xxs'),
            letterSpacing: theme('letterSpacing.xl'),
          }
        },
        '.text-overline': { 
          fontSize: theme('fontSize.3xs'),
          lineHeight: theme('lineHeight.xxs'),
          letterSpacing: theme('letterSpacing.4xl'),
          textTransform: 'uppercase',
          fontWeight: theme('fontWeight.bold'),
          '@screen m': {
            fontSize: theme('fontSize.3xs'),
            lineHeight: theme('lineHeight.xxs'),
            letterSpacing: theme('letterSpacing.4xl'),
          }
        },
        '.divider': { 
          height: '3px',
          '@screen m': {
            height: '4px',
          }
        },
      })
    },
  ]
}
